examples/onnxrt/nlp/huggingface_model/code_detection/quantization/ptq_static/main.py

```diff
@@ -0,0 +1,347 @@
+        inputs_names = [session.get_inputs()[i].name for i in range(len_inputs)]
+        all_labels = []
+        all_preds = []
+        import tqdm
+
+        for idx, (inputs, labels) in tqdm.tqdm(enumerate(dataloader)):
+            if not isinstance(labels, list):
+                labels: List[np.array] = [labels]  # List[shape: bs]
+            inputs = inputs[:len_inputs]
+            for i in range(len_inputs):
+                ort_inputs.update({inputs_names[i]: inputs[i]})
+            predictions: List[np.array] = session.run(
+                None, ort_inputs
+            )  # List[# shape, (bs, 2)]
+            predictions = [np.argmax(p, axis=1) for p in predictions]
+
+            all_labels += labels
+            all_preds += predictions
+            np.mean(
+                np.concatenate(all_labels, 0) == (np.concatenate(all_preds, 0))
+            )  # [:,0]>0.5))
+        label_flatten = np.concatenate(all_labels, 0)
+        preds_flatten = np.concatenate(all_preds, 0)
+        correct_count = np.sum(label_flatten == preds_flatten)
+        acc = correct_count / len(label_flatten)
+        print(acc)
+
+        return acc
+    
+    # tune
+    if args.tune:
+        from neural_compressor import PostTrainingQuantConfig, quantization
+        from onnxruntime.transformers import optimizer
+        from onnxruntime.transformers.fusion_options import FusionOptions
+
+        train_dataset = load_dataset_from_local(args.data_path, args.model_name_or_path)
+        ort_dataset = ONNXRTDataset(args.model_path, train_dataset)
+
+        from neural_compressor.data import DataLoader as INC_DataLoader
+
+        dataloader = INC_DataLoader(
+            framework="onnxruntime", dataset=ort_dataset, batch_size=args.batch_size
+        )
+
+        model_type = "bert"
+        opt_options = FusionOptions(model_type)
+        opt_options.enable_embed_layer_norm = False
+
+        model_optimizer = optimizer.optimize_model(
+            args.model_path,
+            model_type,
+            num_heads=12,
+            hidden_size=768,
+            optimization_options=opt_options,
+        )
+        model = model_optimizer.model
+
+        # check the optimized model is valid
+        import onnxruntime as ort
+
+        try:
+            ort.InferenceSession(
+                model.SerializeToString(), providers=ort.get_available_providers()
+            )
+        except Exception as e:
+            logger.warning("Optimized model is invalid: {}. ".format(e))
+            logger.warning(
+                "Model optimizer will be skipped. "
+                "Try to upgrade onnxruntime to avoid this error"
+            )
+            model = onnx.load(args.model_path)
+
+        config = PostTrainingQuantConfig(
+            approach="static",
+            quant_level=1,
+            quant_format=args.quant_format,
+            # recipes={"smooth_quant": True, "smooth_quant_args": {"alpha": 0.5}},
```

Comment:
need to remove `recipes`?

Reply:
Removed by [0c26397](https://github.com/intel/neural-compressor/pull/1121/commits/0c2639750f33f6b306544dba11175291c4ad691d).